modules/web_hosting/network.tf

```diff
@@ -0,0 +1,184 @@
+  ip_cidr_range            = "10.200.240.0/24"
+  region                   = "asia-south1"
+  network                  = google_compute_network.vpc-xlb.name
+  project                  = local.project.project_id
+  private_ip_google_access = true
+}
+
+#########################################################################
+# Firewall Rules in vpc-xlb
+#########################################################################
+
+# FW rule for L7LB healthcheck
+resource "google_compute_firewall" "fw-vpc-xlb-lb-hc" {
+  project       = local.project.project_id
+  name          = "fw-vpc-xlb-lb-hc"
+  network       = google_compute_network.vpc-xlb.name
+ 
+  allow {
+    protocol    = "tcp"
+    ports       = ["80"]
+  }
+ 
+  source_ranges = ["35.191.0.0/16", "130.211.0.0/22"]
+}
+
+
+# FW rule for ICMP
+resource "google_compute_firewall" "fw-vpc-xlb-allow-icmp" {
+  project       = local.project.project_id
+  name          = "fw-vpc-xlb-allow-icmp"
+  network       = google_compute_network.vpc-xlb.name
+  priority      = 65534
+  allow {
+    protocol    = "icmp"
+  }
+ 
+  source_ranges = ["0.0.0.0/0"]
+}
+
+# FW rule for SSH via IAP
+resource "google_compute_firewall" "fw-vpc-xlb-allow-iap-ssh" {
+  name          = "fw-vpc-xlb-allow-iap-ssh"
+  network       = resource.google_compute_network.vpc-xlb.name
+  project       = local.project.project_id
+  allow {
+    protocol    = "tcp"
+    ports       = ["22"]
+  }
+  source_ranges = ["35.235.240.0/20"]
+}
+
+
+# FW rule for Intra VPC
+resource "google_compute_firewall" "fw-vpc-xlb-allow-intra-vpc" {
+  name          = "fw-vpc-xlb-allow-intra-vpc"
+  network       = resource.google_compute_network.vpc-xlb.name
+  project       = local.project.project_id
+  allow {
+    protocol    = "all"
+  }
+  source_ranges = ["10.100.20.0/24","10.200.240.0/24"]
+}
+
+
+#########################################################################
+# Creating Cloud NATs for Egress traffic from GCE VMs in vpc-xlb
+#########################################################################
+
+resource "google_compute_router" "cr-vpc-xlb-us-c1" {
+  name    = "cr-vpc-xlb-us-c1"
+  project = local.project.project_id
+  region  = google_compute_subnetwork.subnetwork-vpc-xlb-us-c1.region
+  network = google_compute_network.vpc-xlb.id
+
+  bgp {
+    asn = 64514
+  }
+}
+
+resource "google_compute_router_nat" "nat-gw-vpc-xlb-us-c1" {
+  name                               = "nat-gw-vpc-xlb-us-c1"
+  project                            = local.project.project_id
+  router                             = google_compute_router.cr-vpc-xlb-us-c1.name
+  region                             = google_compute_router.cr-vpc-xlb-us-c1.region
+  nat_ip_allocate_option             = "AUTO_ONLY"
+  source_subnetwork_ip_ranges_to_nat = "ALL_SUBNETWORKS_ALL_IP_RANGES"
+
+  log_config {
+    enable = true
+    filter = "ERRORS_ONLY"
+  }
+}
+
+resource "google_compute_router" "cr-vpc-xlb-asia-s1" {
+  name    = "cr-vpc-xlb-asia-s1"
```

Comment:
Any reason why we are deploying this in two regions?

Reply:
We are demonstrating Cross region Load balancing with CDN where one backend is in US and another in ASIA.